src/mysky/index.ts

```diff
@@ -189,11 +189,15 @@ export class MySky {
 
         // Send the UI the list of required permissions.
 
+        // Slightly hacky, but this copies permissions array so that Proxy objects
+        // used in frameworks like Vuex don't try to pass through postMessage.
+        const perms = JSON.parse(JSON.stringify(this.pendingPermissions));
```

Comment:
This looks like something I would do when we're extending the `pendingPermissions`, rather than having `pendingPermissions` contain whatever those Proxy objects are, and doing this hack only in this spot. Wouldn't it be better to apply this hack transform on add?

Reply:
If I understand correctly, I don't think this would actually work. The whole thing is a proxy in a proxy for Vuex, so we don't actually want to try and do this deep-copy for anything that gets saved as part of the state, since Vuex will just write that back to the original object (where it's already correct) and then proxy it in its state.

Reply:
This looks exactly like the explanation Marcin was looking for on line 192. :)